File: src/main/java/Perms.java
package maxdistructo.droidbot2.core;


import sx.blah.discord.handle.obj.*;
import sx.blah.discord.util.PermissionUtils;

import static maxdistructo.droidbot2.core.Client.client;

public class Perms {
    public static boolean checkMod(IMessage message){
        IUser author = message.getAuthor();
        //Checks if user is a Moderator of the server
        long[] moderators = Config.readServerModConfig(message.getGuild());
        int i = 0;
        while(i < moderators.length){
            if(author.getLongID() == moderators[i] || checkAdmin(message) ||  checkOwner_Guild(message)){
                return true;
            }
            i++;
        }
        return false;
    }

    public static boolean checkAdmin(IMessage message){
        IUser author = message.getAuthor();
        //Checks if user is a Admin/Owner of the Server (Or Myself).
        long[] admins = Config.readServerAdminConfig(message.getGuild());
        int i = 0;
        while(i < admins.length){
            if(author.getLongID() == admins[i] || checkOwner_Guild(message)){
                return true;
            }
            i++;
        }
        return false;
    }

    public static boolean checkOwner_Guild(IMessage message){
        IUser author = message.getAuthor();

        return author.getLongID() == message.getGuild().getOwnerLongID()|| checkOwner(message);
    }
    public static boolean checkOwner(IMessage message){
       IUser author = message.getAuthor();
        return author.getLongID() == client.getApplicationOwner().getLongID() || author.getLongID() == 374517920505790464L;
    }

    public static boolean checkGames(IMessage message){
        IChannel channel = message.getChannel();
        String channelName = channel.getName();

        String[] channels = Config.readServerGamesConfig(message.getGuild());
        int i = 0;
        while(i < channels.length){
            if(channelName.equals(channels[i])){
                return true;
            }
            i++;
        }
        return false;
    }
    public static boolean checkForPermission(IMessage message, Permissions permission){
        return PermissionUtils.hasPermissions(message.getGuild(), message.getAuthor(), permission);
    }
}

File: src/main/java/maxdistructo/droidbot2/core/Perms.kt
package maxdistructo.droidbot2.core


import sx.blah.discord.handle.obj.*
import sx.blah.discord.util.PermissionUtils

import maxdistructo.droidbot2.core.Client.client

object Perms {
    fun checkMod(message: IMessage): Boolean {
        val author = message.author
        //Checks if user is a Moderator of the server
        val moderators = Config.readServerModConfig(message.guild)
        var i = 0
        while (i < moderators.size) {
            if (author.longID == moderators[i] || checkAdmin(message) || checkOwner_Guild(message)) {
                return true
            }
            i++
        }
        return false
    }

    fun checkAdmin(message: IMessage): Boolean {
        val author = message.author
        //Checks if user is a Admin/Owner of the Server (Or Myself).
        val admins = Config.readServerAdminConfig(message.guild)
        var i = 0
        while (i < admins.size) {
            if (author.longID == admins[i] || checkOwner_Guild(message)) {
                return true
            }
            i++
        }
        return false
    }

    fun checkOwner_Guild(message: IMessage): Boolean {
        val author = message.author

        return author.longID == message.guild.ownerLongID || checkOwner(message)
    }

    fun checkOwner(message: IMessage): Boolean {
        val author = message.author
        return author.longID == client!!.applicationOwner.longID || author.longID == 374517920505790464L
    }

    fun checkGames(message: IMessage): Boolean {
        val channel = message.channel
        val channelName = channel.name

        val channels = Config.readServerGamesConfig(message.guild)
        var i = 0
        while (i < channels.size) {
            if (channelName == channels[i]) {
                return true
            }
            i++
        }
        return false
    }

    fun checkForPermission(message: IMessage, permission: Permissions): Boolean {
        return PermissionUtils.hasPermissions(message.guild, message.author, permission)
    }
}
